src/sentry/middleware/devtoolbar.py

```diff
@@ -0,0 +1,70 @@
+import logging
+
+from django.http import HttpRequest, HttpResponse
+
+from sentry import analytics
+from sentry.utils.http import origin_from_request
+from sentry.utils.http import query_string as get_query_string
+from sentry.utils.urls import parse_id_or_slug_param
+
+logger = logging.getLogger(__name__)
+
+
+class DevToolbarAnalyticsMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        try:
+            response = self.get_response(request)
+        except Exception:
+            response = HttpResponse(status=500)
+            raise  # delegate to our API exception handlers
+        finally:
+            _record_api_request(request, response)
+
+        return response
```

Comment:
it's unclear to me that we need this. i'd rather not mess with our default exception logic, which i believe this would do. if the endpoint 500s we do not need to worry about capturing analytics logs. https://github.com/getsentry/sentry/blob/eeedd4c28fe73323fca4352ef4db77c1ed1b1b67/src/sentry/middleware/flag.py#L13 can we make it similar to this?

Reply:
I thought it'd be useful to see when toolbar requests are resulting in 500s. It does seem like an edge case tho, @ryan953 do you have thoughts?

Reply:
> I thought it'd be useful to see when toolbar requests are resulting in 500s. 

not worth modifying our complicated-as-is exception waterfall IMO. i think errors are actually caught before this is even ran, so i don't think this would do anything.

you should be able to look at referrers from the sentry issue.

Reply:
Sounds good!